File: src/main/java/me/openani/commands/HelpCommand.java
package me.openani.commands;

import org.json.JSONObject;
import me.openani.handler.command.Command;
import me.openani.handler.command.CommandContext;
import me.openani.handler.listener.CommandListener;

public class HelpCommand implements CommandListener {
    @Override
    public void onCommand(CommandContext context) {
        JSONObject commands = new JSONObject();

        for (Command command : context.getCommandHandler().getCommands()) {
            commands.put(command.getName(), command.getDescription());
        }

        String help = "```json\n" + commands.toString(4) + "```";

        context.channel.sendMessage(help).queue();
    }
}

File: src/main/kotlin/me/openani/commands/HelpCommand.kt
package me.openani.commands

import org.json.JSONObject
import me.openani.handler.command.CommandContext
import me.openani.handler.listener.CommandListener

class HelpCommand: CommandListener {
    override fun onCommand(context: CommandContext) {
        val commands = JSONObject()

        for (cmd in context.commandHandler.commands) {
            commands.put(cmd.name, if (cmd.description.isNullOrEmpty()) "None" else cmd.description)
        }

        val result = "```json\n${commands.toString(4)}```"

        context.channel.sendMessage(result).queue()
    }
}